client/SMCE_Client.cpp

```diff
@@ -0,0 +1,339 @@
+#include <SMCE/Sketch.hpp>                                                                                                       
+#include <SMCE/SketchConf.hpp>                                                                                                   
+#include <SMCE/Toolchain.hpp>
+                                                                                               
+                                                                                                                                 
+using namespace std::literals;                                                                                                   
+                                                                                                                                                                                                                                                                                                    
+void print_help(const char* argv0){                                                                                              
+    std::cout << "Usage: " << argv0 << " <fully-qualified-board-name> <path-to-sketch>" << std::endl;                            
+}                                                                                                                                
+                                                                                                                                 
+void exit_sketch(int code){                                                                                                      
+    std::cerr << "Error code:  "<< code << std::endl;                                                                            
+}                                                                                                                                
+
+// atomic bool for handling threads
+static std::atomic_bool run_threads = true;
+static std::atomic_bool mute_uart = false;
+static std::mutex t_lock;
+//Listen to the uart input from board, writes what it read to terminal                                                           
+void uart_listener(smce::VirtualUart uart,bool file_write,std::string path){                                                                                      
+    auto tx = uart.tx();
+    std::string file = path+"/uartlog"+get_time()+".txt";                                                                                                         
+    while(run_threads){                                                                                                          
+        std::string buffer;
+        t_lock.lock(); // if recompiling sketch, lock thread here until recompiling is done.
+        buffer.resize(tx.max_size());                                                                                            
```

Comment:
The max size is an invariant (once it has been initialized), so it would be wiser to reserve the space once, before the loop starts, so that you do not spend time allocating/releasing memory every time you have new data to read